### DISCUSSION TITLE
Generating some form of bitfield from constants

### DISCUSSION THREAD

**Discussion by belak** [rust-lang/rust-bindgen#2515] [Q&A] [answered]
I have a bunch of constants, like this:

```
#define SDL_HWACCEL 0x00000100
#define SDL_SRCCOLORKEY 0x00001000
#define SDL_RLEACCELOK 0x00002000
#define SDL_RLEACCEL 0x00004000
#define SDL_SRCALPHA 0x00010000
#define SDL_PREALLOC 0x01000000
```

I believe they can be OR'd together to create a final value - is there a way to create a rust type which will represent that, or will it need to be done manually?

**Comment by pvdrz** [comment, marked as answer]
That's a bit out of the scope of bindgen. You could use something like the `bitflags` crate on top of it I suppose.

**Reply by belak** [reply]
That makes sense - probably not something you'd include in a *-sys library anyway. Thanks for the pointer!
